### PR TITLE
Use Yarn Workspaces

### PR DESCRIPTION
**Pull request by hzoo**
https://twitter.com/left_pad/status/893440394427600896

- [x] need to update yarn on CI
- [x] updated lerna-changelog (don't need the repo property anymore)
- [x] using yarn workspaces
- [x] had to use an older version of babel since otherwise it would be the same version in devDeps and in packages/ and thus it symlinks to itself 😛 

```
make bootstrap  20.76s user 7.92s system 132% cpu 21.697 total
make bootstrap  141.20s user 27.25s system 366% cpu 45.976 total
```

basically 2x faster?

### CODE REVIEW

**Comment by hzoo** [comment]
Ok there is a weird

```
warning The case-insensitive file /home/travis/build/babel/babel/node_modules/nyc/node_modules shouldn't be copied twice in one bulk copy
```
warning (there is a fix in yarn but not released yet)

Also had to uncomment flow since our code is just not tested with it at all at this point.

File: lerna.json [hunk 1]
```diff
@@ -21,5 +20,10 @@
         "test/**"
       ]
     }
-  }
+  },
+  "packages": [
+    "packages/*"
```
Comment (Hypnosphi) [inline lerna.json]: Why is this needed? Lerna should use workspaces field from `package.json` when `useWorkspaces` flag is passed

File: package.json [hunk 2]
```diff
@@ -28,16 +28,16 @@
     "eslint-config-babel": "^7.0.1",
     "eslint-plugin-flowtype": "^2.20.0",
     "eslint-plugin-prettier": "^2.1.2",
-    "flow-bin": "^0.41.0",
+    "flow-bin": "^0.52.0",
     "gulp": "^3.9.0",
-    "gulp-babel": "7.0.0-alpha.18",
+    "gulp-babel": "babel/gulp-babel#d8c67b",
```
Comment (styfle) [inline package.json]: What does this syntax do? I've never seen it before 😄

Comment (justin-cruz-ck) [reply package.json]: This targets a specific commit, in this case https://github.com/babel/gulp-babel/commit/d8c67b

Comment (Digikid13) [reply package.json]: This will pick a specific commit, in this case: https://github.com/babel/gulp-babel/commit/d8c67b

File: lerna.json (hunk 1 above)
Comment (ravicious) [reply lerna.json]: This is here because this PR is from last month when Lerna didn't support this yet AFAIR.
